src/redux/books/books.js

```diff
@@ -1,32 +1,11 @@
 import { createAction } from '@reduxjs/toolkit';
+import api from '../../api/api';
 
 export const ADD_BOOK = createAction('ADD_BOOK');
 export const REMOVE_BOOK = createAction('REMOVE_BOOK');
+export const GET_BOOKS = createAction('GET_BOOKS');
 
-const initialdata = [
-  {
-    id: '1',
-    title: 'java',
-    author: 'Rashid',
-  },
-  {
-    id: '2',
-    title: 'javascript',
-    author: 'Menhas',
-  },
-  {
-    id: '3',
-    title: 'Data structure and Algorithm ',
-    author: 'Sami ullah',
-  },
-  {
-    id: '4',
-    title: ' Algorithm ',
-    author: 'Saboor',
-  },
-];
-
-export const bookReducer = (state = initialdata, action) => {
+export const bookReducer = (state = [], action) => {
```

Comment:
Hello  @RashidMenhas as previously requested kindly use createAsyncThunk as it is  a project requirement.
 Follow the example given below:

export const getBooks = createAsyncThunk(
  'books/getBooks',
  async () => {
    const response = await axios.get(
      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/',
    );
    return response.data;
  },